packages/@aws-cdk/aws-ecs/lib/ec2/ec2-task-event-rule-target.ts

```diff
@@ -0,0 +1,101 @@
+import events = require ('@aws-cdk/aws-events');
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+import { TaskDefinition } from '../base/task-definition';
+import { ICluster } from '../cluster';
+import { isEc2Compatible } from '../util';
+
+/**
+ * Properties to define an EC2 Event Task
+ */
+export interface Ec2EventTaskProps {
+  /**
+   * Cluster where service will be deployed
+   */
+  cluster: ICluster;
+
+  /**
+   * Task Definition of the task that should be started
+   */
+  taskDefinition: TaskDefinition;
+
+  /**
+   * How many tasks should be started when this event is triggered
+   *
+   * @default 1
+   */
+  taskCount?: number;
+}
+
+/**
+ * Start a service on an EC2 cluster
+ */
+export class Ec2TaskEventRuleTarget extends cdk.Construct implements events.IEventRuleTarget {
+  private readonly cluster: ICluster;
+  private readonly taskDefinition: TaskDefinition;
+  private readonly taskCount: number;
+
+  constructor(scope: cdk.Construct, id: string, props: Ec2EventTaskProps) {
+    super(scope, id);
+
+    if (!isEc2Compatible(props.taskDefinition.compatibility)) {
+      throw new Error('Supplied TaskDefinition is not configured for compatibility with EC2');
+    }
+
+    this.cluster = props.cluster;
+    this.taskDefinition = props.taskDefinition;
+    this.taskCount = props.taskCount !== undefined ? props.taskCount : 1;
+  }
+
+  /**
+   * Allows using containers as target of CloudWatch events
+   */
+  public asEventRuleTarget(_ruleArn: string, _ruleUniqueId: string): events.EventRuleTargetProps {
+    const role = this.eventsRole;
+
+    role.addToPolicy(new iam.PolicyStatement()
+      .addAction('ecs:RunTask')
+      .addResource(this.taskDefinition.taskDefinitionArn)
+      .addCondition('ArnEquals', { "ecs:cluster": this.cluster.clusterArn }));
+
+    return {
+      id: this.node.id,
+      arn: this.cluster.clusterArn,
+      roleArn: role.roleArn,
+      ecsParameters: {
+        taskCount: this.taskCount,
+        taskDefinitionArn: this.taskDefinition.taskDefinitionArn
+      }
+    };
+  }
+
+  /**
+   * Create or get the IAM Role used to start this Task Definition.
+   *
+   * We create it under the TaskDefinition object so that if we have multiple EventTargets
+   * they can reuse the same role.
+   */
+  public get eventsRole(): iam.IRole {
+    let role = this.taskDefinition.node.tryFindChild('EventsRole') as iam.IRole;
```

Comment:
Should taskDefinition expose the role as a public property?

Reply:
I went back and forth on that. Ultimately decided that it was nice for TaskDefinition to not have to care about any of this.

Reply:
I like this better. The concern about the role should be kept here in my mind.
I would use a more unique name (uuid?) to ensure it doesn't conflict with something a user would add